racksdb/web/ui/src/components/ComboBox.vue

```diff
@@ -0,0 +1,127 @@
+<!--Copyright (c) 2022-2024 Rackslab
+
+This file is part of RacksDB.
+
+SPDX-License-Identifier: GPL-3.0-or-later -->
+
+<script setup lang="ts">
+import { ref, computed, watchEffect, onMounted } from 'vue'
+import type { PropType } from 'vue'
+import type { Ref } from 'vue'
+import { useRouter } from 'vue-router'
+import type { Datacenter, Infrastructure } from '@/composables/RacksDBAPI'
+import {
+  Combobox,
+  ComboboxInput,
+  ComboboxOption,
+  ComboboxOptions,
+  ComboboxLabel
+} from '@headlessui/vue'
+import { CheckIcon } from '@heroicons/vue/24/outline'
+
+const selectedItem: Ref<string | undefined> = ref()
+const router = useRouter()
+const input = ref('')
+const inputComponent: Ref<typeof ComboboxInput | null> = ref(null)
+const filteredItems = computed(() =>
+  input.value === ''
+    ? props.items
+    : props.items.filter((item) => {
+        return item.name.toLocaleLowerCase().includes(input.value.toLocaleLowerCase())
+      })
+)
+
+function goToItem(item: string) {
+  router.push({
+    name: props.itemType + 'details',
+    params: { name: item }
+  })
+}
+
+const props = defineProps({
+  itemType: {
+    type: String as PropType<'datacenter' | 'infrastructure'>,
+    required: true
+  },
+  items: {
+    type: Array<Datacenter | Infrastructure>,
+    default: []
+  }
+})
+
+/* 
+  watchEffect detects any change in selectedItem and call the function goToItem()
+  and pass the change in parameter
+*/
```

Comment:
This comment describes the code as it can be read but it does not tell the _purpose_ of this code. You should explain the purpose (ie. why is it here?).

Reply:
I updated the comment to explain that after detecting a change, it triggers a function that redirect the user to the desired view.